An federation whitelist query endpoint extension (#16848)
element-hq/synapse: docs/usage/configuration/config_documentation.md

  - syd.example.com
```
---
### `federation_whitelist_endpoint_enabled`

Enables an endpoint for fetching the federation whitelist config.

The request method and path is `GET /_synapse/client/config/federation_whitelist`, and the

> response format is:

Reviewer: Sorry, last minute change. We should be versioning this endpoint, in case we need to make a change to it in the future.

```
GET /_synapse/client/v1/config/federation_whitelist
```

Reviewer 2: Indeed, good catch. Will do

    min_batch_size: 10
    default_batch_size: 50
```

---
## Extension features
Configuration for extension features for Synapse

---
### `extension_federation_whitelist_endpoint`

Enables an endpoint for fetching the federation whitelist config.

> The request path is `/_synapse/client/config/federation_whitelist`, and the

Reviewer: ```suggestion
The request method and path is `GET /_synapse/client/config/federation_whitelist`, and the
```

response format is:

```json
{

>     "whitelist_enabled": true,  // Whether there is a federation whitelist
>     "whitelist": [  // Which hosts are allowed by the whitelist

Reviewer: ```suggestion
    "whitelist_enabled": true,  // Whether the federation whitelist is being enforced
    "whitelist": [  // Which server names are allowed by the whitelist
```

likewise in the docstring of `FederationWhitelistResource`.

        "example.com"
    ]
}
```

> If `whitelist_enabled` is `false` then the server can federate with all others.

Reviewer: nit

```suggestion
If `whitelist_enabled` is `false` then the server is permitted to federate with all others.
```
